### PR TITLE
[IMP] account: allow drag and drop to upload bills in list view

### PR DESCRIPTION
**Pull request by william-andre**
Make users feel at home with a warm welcome when they try to upload
something with a drag and drop.

[task-2748604](https://www.odoo.com/web#id=2748604&model=project.task)

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by csnauwaert — CHANGES_REQUESTED** [review]
Community review, found a traceback and a strange behavior while testing :)

File: addons/account/static/src/js/bills_upload.js [hunk 1]
```diff
@@ -36,114 +56,106 @@ odoo.define('account.upload.bill.mixin', function (require) {
...
         }),
     });
-
-    var BillsKanbanView = KanbanView.extend({
+        
+    var DocumentKanbanView = KanbanView.extend({
         config: _.extend({}, KanbanView.prototype.config, {
-            Controller: BillsKanbanController,
+            Controller: KanbanController.extend({_dropZone: '.o_content'}, UploadDocumentMixin, {
+                buttons_template: 'DocumentKanbanView.buttons',
```
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/static/src/js/bills_upload.js:121]: Changed everything to `DocumentsKanbanView` for more consistency

File: addons/account/static/src/js/bills_upload.js [hunk 2]
```diff
@@ -36,114 +56,106 @@ odoo.define('account.upload.bill.mixin', function (require) {
...
+                events: _.extend({}, KanbanController.prototype.events, UploadDocumentMixin.events),
```
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/static/src/js/bills_upload.js:122]: Changing to `Object.assign`, but FYI spread was the slowest in my case
image

File: addons/account/static/src/js/bills_upload.js [hunk 3]
```diff
@@ -36,114 +56,106 @@ odoo.define('account.upload.bill.mixin', function (require) {
...
-                this.initialState.context['default_move_type'] = 'in_invoice'
-            }
-            UploadBillMixin._onUpload.apply(this, arguments);
-        }
+    
+    
+    var DashboardKanbanController = KanbanController.extend({_dropZone: '.o_kanban_record'}, UploadDocumentMixin, {
+        events: _.extend({}, KanbanController.prototype.events, UploadDocumentMixin.events),
+
+         _setJournalContext: function(ev) {
```
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/static/src/js/bills_upload.js]: > don't really like the way we go fetch the id of the journal

Note that this was already like this before, but I changed it to get the value from `kanbanRecord` instead

> Also another problem is that the upload button only works first time

Seems to be working now :shrug:

File: addons/account/static/src/js/bills_upload.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (csnauwaert) [reply addons/account/static/src/js/bills_upload.js]: Funny :smile: , might have something to do with the browser as we should have similar computers. Tried on two different laptops and results were consistent for me, using latest version of chrome though

**Comment by csnauwaert** [comment]
@robodoo r+

**Pull request closed without merging**